Sources/XiEditor/MarkerBar.swift

```diff
@@ -32,17 +32,17 @@ class Marker {
 class MarkerBar: NSScroller {
     var markers: [Marker] = []
     var backgroundColor: NSColor = NSColor.red
-    var parent: EditViewController!
-    var markerLayer = CAShapeLayer()
-    var overlayScrollerLayer: CALayer?
+    let markerLayer = CAShapeLayer()
+    var overlayScrollerLayer: CALayer!
```

Comment:
implicitly unwrapped optionals are dangerous, and should be used with caution. Basically: you should only use an implicitly unwrapped optional for things that are guaranteed to be present during normal use,  but which can't be present when the object is init'd; a good example of this is some subview that is always created in `viewDidLoad`, and exists for the rest of the object's lifecycle, but can't exist in `init`. Just making this an optional makes the intent clearer.